### ISSUE TITLE
Stdout buffer flush in Write String function

### ISSUE THREAD

**Issue by felipefoz** [G-CLI/G-CLI#177]
Hey,

I noticed a behavior of the write to string function that is not what I expected. After some investigation I figured out that the Stdout by default is buffered, and depending on the configuration will only flush its outputs if there is an end of line, which is what I understand the rust part of g-cli is doing.
So, everytime I use the write string function, nothing happens, and only when a write line function is called, the terminal is populated.

My use case is related to writing pieces of string for a progress for example, so user can see the progress, without resorting to use the new line.

Python for example, has this sys.stdout.flush(), rust I think is std::io::stdout().flush()

Could we incorporate in G-CLI an option in the "Write String" function, where we could set a flag to force flushing the Stdout buffer?

Regards,

**Comment by JamesMc86** [comment]
Sounds like a great change. I would probably be inclined to make it part of the LabVIEW side so it is similar to having the APIs native there but will look through some of the options

**Comment by felipefoz** [comment]
Thanks for the feedback. 
I am in doubt actually if this can be done from LabVIEW side, as this is a behavior from the application writing to terminal.
I don't know how big the change would be (no rust expert), but I guess it would need to create a new message type, and handing that specific one to write immediately as the others should continue to be the same.

Right now I have a workaround with escape codes, it is clunky and not ideal but it works.

**Comment by JamesMc86** [comment]
You are absolutely correct - I've confirmed that the flush needs to happen on the Rust side.

My inclination is to give the LabVIEW user control over this so I'll add a new message and API for this.

**Comment by felipefoz** [comment]
Sounds good. Let me know if you need any help on testing whenever you come to this.

**Comment by JamesWiresmith** [comment]
I have a version of this in now. I'm hoping to work on a couple more bits at the end of the week and can create a built to test with.
